### ISSUE TITLE
auth: Support for catalog zones custom properties

### ISSUE THREAD

**Issue by rgacogne** [PowerDNS/pdns#13682]
- Program: Authoritative
 - Issue type: Feature request

### Short description

BIND defines custom properties for ACL, primaries and TSIG settings: https://bind9.readthedocs.io/en/stable/chapter6.html#catalog-zone-custom-properties and I would like a way to do the same with PowerDNS.

### Usecase
My use-case is simple: I want to provision the TSIG keys and ACL to use on the secondaries for the zones learned on the consumer via the catalog zone itself.

I gave it a shot in https://github.com/PowerDNS/pdns/pull/12772 but my implementation did not match the current design.

**Comment by insertjokehere** [comment]
This is something I am also interested in. In my setup, all zones require TSIG keys to AXFR - I can get a secondary to transfer the catalog zone by setting `AXFR-MASTER-TSIG` as normal, but then the actual zones don't transfer because PowerDNS doesn't use the TSIG key.

I can work around it by adding the secondaries' IP to `allow-axfr-ips` on the primary, but this seems not ideal. Did you find any other way to get zones provisioned through a catalog zone to get the settings you were after?

**Comment by rgacogne** [comment]
Not really, I have an ugly solution where a timer is regularly checking all zones to check if the expected meta-data is set, and setting it otherwise. It works well for my use-case but it's really sad to have to do this out-of-band.

**Comment by gaoyifan** [comment]
I tried another sidecar approach, listening to the PowerDNS logs for matches like `Catalog-Zone create zone`, and activating TSIG for the new zone. It works, but ulgy.
